src/main/java/com/google/api/generator/gapic/composer/ServiceClientSampleCodeComposer.java

```diff
@@ -194,121 +194,166 @@ public static String composeRpcMethodHeaderSampleCode(
+    bodyExprs.addAll(rpcMethodArgAssignmentExprs);
+
+    List<Statement> bodyStatements = new ArrayList<>();
     if (method.isPaged()) {
-      // Find the repeated field.
-      Message methodOutputMessage = messageTypes.get(method.outputType().reference().simpleName());
-      Field repeatedPagedResultsField = methodOutputMessage.findAndUnwrapFirstRepeatedField();
-      Preconditions.checkNotNull(
-          repeatedPagedResultsField,
-          String.format(
-              "No repeated field found on message %s for method %s",
-              methodOutputMessage.name(), method.name()));
-
-      TypeNode repeatedResponseType = repeatedPagedResultsField.type();
-      return SampleCodeWriter.write(
-          composeUnaryPagedRpcMethodSampleCode(
-              method, clientType, repeatedResponseType, arguments, resourceNames));
-    }
-    if (method.hasLro()) {
-      return SampleCodeWriter.write(
-          composeUnaryLroRpcMethodSampleCode(method, clientType, arguments, resourceNames));
+      bodyStatements.addAll(
+          composeUnaryPagedRpcMethodSampleCodeBodyStatements(
+              method, clientVarExpr, rpcMethodArgVarExprs, bodyExprs, messageTypes));
+    } else if (method.hasLro()) {
+      bodyStatements.addAll(
+          composeUnaryLroRpcMethodSampleCodeBodyStatements(
+              method, clientVarExpr, rpcMethodArgVarExprs, bodyExprs));
+    } else {
+      bodyStatements.addAll(
+          composeUnaryRpcMethodSampleCodeBodyStatements(
+              method, clientVarExpr, rpcMethodArgVarExprs, bodyExprs));
     }
+
     return SampleCodeWriter.write(
-        composeUnaryRpcMethodSampleCode(method, clientType, arguments, resourceNames));
+        TryCatchStatement.builder()
+            .setTryResourceExpr(assignClientVariableWithCreateMethodExpr(clientVarExpr))
+            .setTryBody(bodyStatements)
+            .setIsSampleCode(true)
+            .build());
   }
 
-  @VisibleForTesting
-  static TryCatchStatement composeUnaryRpcMethodSampleCode(
+  public static String composeRpcDefaultMethodHeaderSampleCode(
       Method method,
       TypeNode clientType,
-      List<MethodArgument> arguments,
-      Map<String, ResourceName> resourceNames) {
+      Map<String, ResourceName> resourceNames,
+      Map<String, Message> messageTypes) {
     VariableExpr clientVarExpr =
         VariableExpr.withVariable(
             Variable.builder()
                 .setName(JavaStyle.toLowerCamelCase(clientType.reference().name()))
                 .setType(clientType)
                 .build());
-    List<VariableExpr> rpcMethodArgVarExprs = createRpcMethodArgumentVariableExprs(arguments);
-    List<Expr> rpcMethodArgDefaultValueExprs =
-        createRpcMethodArgumentDefaultValueExprs(arguments, resourceNames);
-    List<Expr> bodyExprs =
-        createAssignmentsForVarExprsWithValueExprs(
-            rpcMethodArgVarExprs, rpcMethodArgDefaultValueExprs);
+
+    List<Expr> bodyExprs = new ArrayList<>();
```

Comment:
Nit: Move this to just before the `bodyExprs.add()` invocation.

Reply:
Done.